drivers/caldav/caldav_driver.php

```diff
@@ -101,9 +101,41 @@ public function __construct($cal)
         if(self::$debug === null)
             self::$debug = $this->rc->config->get('calendar_caldav_debug', False);
 
+        $this->_setup_preinstalled_sources();
         $this->_read_calendars();
     }
 
+    /**
+     * Setup preinstalled sources defined in config file
+     */
+    protected function _setup_preinstalled_sources()
+    {
```

Comment:
Sorry for being picky. One last thought I had: This will be called everytime a request is sent to the server which happens quite a lot. Since it potentially involves a lot of database calls, we should make sure it only gets called once.

So I suggest adding something like this (not tested):
```
if($this->rc->config->get('calendar_caldav_preinstalled_sources_set', false)) {
    return;
}
$this->rc->config->set('calendar_caldav_preinstalled_sources_set', true);
```

Reply:
I understand why you would prefer this. But one thing I like with how it works currently is that even if someone deletes the preinstalled source, it will be automatically recreated.

If this is done like you are suggesting it, I would like to be able to define a source as "non-removable".

Also, would the code that you are suggesting work on a per-user basis ?

Reply:
Hmm. I see your point. Sorry I didnt even think about that.
I honestly cant think of any better solution. So lets just roll with it :)

Thank you very much for your effort and time!